### PR TITLE
IoT: Added Rules implementation

### PR DESCRIPTION
**Pull request by sim1234**

### CODE REVIEW

File: tests/test_iot/test_iot.py [hunk 1]
```diff
@@ -1991,3 +1991,133 @@ def test_list_job_executions_for_thing():
...
+    # errors
+
+    # get not existing rule
+    try:
+        client.get_topic_rule(ruleName=name)
+    except ClientError as exc:
+        error_code = exc.response["Error"]["Code"]
+        error_code.should.equal("ResourceNotFoundException")
+    else:
+        raise Exception("Should have raised error")
```
Comment (bblommers) [inline tests/test_iot/test_iot.py:2077]: Error handling should follow this format:
```
with pytest.raises(..):
    ....
````
Like it's done in an earlier test case: https://github.com/spulec/moto/blob/master/tests/test_iot/test_iot.py#L607

**Review by bblommers — CHANGES_REQUESTED** [review]
Thanks for the PR @sim1234! Would you mind splitting up the tests into multiple test cases? Ideally each method should only test a single functionality.

File: tests/test_iot/test_iot.py (hunk 1 above)
Comment (sim1234) [reply tests/test_iot/test_iot.py]: @bblommers I've updated the tests implementation. Please check it out.

**Review by bblommers — CHANGES_REQUESTED** [review]
Looks like this is failing in Python2:
https://travis-ci.org/github/spulec/moto/jobs/754133587

Date formatting is usually done like this in Moto, to ensure it works in both Py2 and Py3:
`datetime.datetime.utcnow().strftime("%Y-%m-%d %H:%M:%S")`

**Review by bblommers — APPROVED** [review]
LGTM. Thanks again, @sim1234!

**Comment by bblommers** [comment]
This is now part of moto >= 1.3.16.dev201

**Pull request merged**